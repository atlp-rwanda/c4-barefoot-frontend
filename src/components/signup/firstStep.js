import React from 'react';
import PropTypes from 'prop-types';
import { Formik, Form, Field } from 'formik';
import {TextField, InputAdornment, Box, Avatar} from '@material-ui/core';
import { AccountCircle, Email, Lock } from "@material-ui/icons";
import { makeStyles } from '@material-ui/core/styles';
import { Button } from '@material-ui/core';
import * as yup from 'yup';
import { Link, BrowserRouter as Router, Route } from 'react-router-dom';
import Skeleton from '@material-ui/lab/Skeleton';
import { Login } from '../views/Login';

const useStyles = makeStyles(theme => ({
  form: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    '@media(max-width: 320px)' : {
      display: 'unset'
    },
  },
  button: {
    margin: theme.spacing(1)
  },
  stepButtons: {
    display: 'grid',
    gridTemplateColumns: ' auto 1fr auto',
    alignItems: 'center',
    '@media(max-width: 740px)' : {
      display: 'flex',
      flexDirection: 'row',
      justifyContent: 'space-around'
    },
  },
  box:{
    display: 'flex',
    alignItems: 'center',
    height: '48px',
    marginTop: '16px',
    marginBottom: '8px',
  },
  textBox: {
    height: '38px',
    marginLeft: '0.5rem',
    width: '100%',
  },
  skeletonButton: {
    width: '68px',
    height: '38px',
    borderRadius: '5px'
  },
  skeletonLink: {
    width: '120px',
    borderRadius: '5px'
  }
}));

const validationSchema = yup.object({
    first_name: yup
        .string()
        .required('First Name is required')
        .max(20),
    last_name: yup
        .string()
        .required('Last Name is required')
        .max(20),
    username: yup
        .string()
        .required('Username is required')
        .max(20),
    email: yup
        .string()
        .email('Invalid email')
        .required('Email is required'),
    password: yup
        .string()
        .required("Password is required")
        .min(8),
    confirmPassword: yup
        .string()
        .required("Confirm Password")
        .when("password", {
        is: val => (val && val.length > 0 ? true : false),
        then: yup.string().oneOf(
            [yup.ref("password")],
            "Passwords must be equel"
        )
    })

});

export const FirstStep = ({ loading, formData, setFormData, nextStep }) => {
  const classes = useStyles();
  return (
    <>
      <Formik
        initialValues={formData}
        onSubmit={values => {
          setFormData(values);
          nextStep();
        }}
        validationSchema={validationSchema}
      >
        {({ errors, touched }) => (
          <Form className={classes.form} form-data='form-1'>
            {loading ? (
              <Box className={classes.box}>
                <Skeleton variant="circle">
                  <Avatar/>
                </Skeleton>
                <Skeleton variant="rect" className={classes.textBox} />
              </Box>
              
            ) : (
              <Field
                name='first_name'
                label='First Name *'
                margin='normal'
                as={TextField}
                error={touched.first_name && errors.first_name}
                helperText={touched.first_name && errors.first_name}
                InputProps={{
                  startAdornment: (
                    <InputAdornment position="start">
                      <AccountCircle />
                    </InputAdornment>
                  ),
                }}
              />
            )}
            {loading ? (
              <Box className={classes.box}>
              <Skeleton variant="circle">
                <Avatar/>
              </Skeleton>
              <Skeleton variant="rect" className={classes.textBox} />
            </Box>
            ) : (
              <Field
                name='last_name'
                label='Last Name *'
                margin='normal'
                as={TextField}
                error={touched.last_name && errors.last_name}
                helperText={touched.last_name && errors.last_name}
                InputProps={{
                  startAdornment: (
                    <InputAdornment position="start">
                      <AccountCircle />
                    </InputAdornment>
                  ),
                }}
              />
            )}
            {loading ? (
              <Box className={classes.box}>
                <Skeleton variant="circle">
                  <Avatar/>
                </Skeleton>
                <Skeleton variant="rect" className={classes.textBox} />
              </Box>
            ) : (
              <Field
                name='username'
                label='Username *'
                margin='normal'
                as={TextField}
                error={touched.username && errors.username}
                helperText={touched.username && errors.username}
                InputProps={{
                  startAdornment: (
                    <InputAdornment position="start">
                      <AccountCircle />
                    </InputAdornment>
                  ),
                }}
              />
            )}
            
            {loading ? (
              <Box className={classes.box}>
                <Skeleton variant="circle">
                  <Avatar/>
                </Skeleton>
                <Skeleton variant="rect" className={classes.textBox} />
              </Box>
            ) : (
              <Field
                type='email'
                name='email'
                label='Email *'
                margin='normal'
                as={TextField}
                error={touched.email && errors.email}
                helperText={touched.email && errors.email}
                InputProps={{
                  startAdornment: (
                    <InputAdornment position="start">
                      <Email />
                    </InputAdornment>
                  ),
                }}
              />
            )}
            
            {loading ? (
              <Box className={classes.box}>
                <Skeleton variant="circle">
                  <Avatar/>
                </Skeleton>
                <Skeleton variant="rect" className={classes.textBox} />
              </Box>
            ) : (
              <Field
                type='password'
                name='password'
                label='Password *'
                margin='normal'
                as={TextField}
                error={touched.password && errors.password}
                helperText={touched.password && errors.password}
                InputProps={{
                  startAdornment: (
                    <InputAdornment position="start">
                      <Lock />
                    </InputAdornment>
                  ),
                }}
              />
            )}
            
            {loading ? (
              <Box className={classes.box}>
                <Skeleton variant="circle">
                  <Avatar/>
                </Skeleton>
                <Skeleton variant="rect" className={classes.textBox} />
              </Box>
            ) : (
              <Field
                type='password'
                name='confirmPassword'
                label='Confirm Password *'
                margin='normal'
                as={TextField}
                error={touched.confirmPassword && errors.confirmPassword}
                helperText={touched.confirmPassword && errors.confirmPassword}
                InputProps={{
                  startAdornment: (
                    <InputAdornment position="start">
                      <Lock />
                    </InputAdornment>
                  ),
                }}
              />
            )}
            
            <div className={classes.stepButtons}>
              {loading ? (
                <Box className={classes.box}>
                  <Skeleton variant="rect" className={classes.skeletonLink} />
                </Box>
              ) : (
                <a href='/login' style={{textDecoration:'none'}}>Sign in instead</a>
              )}
              
              <div></div>
              { loading ? (
              <Box className={classes.box}>
                <Skeleton variant="rect" className={classes.skeletonButton} />
              </Box>
              
            ) : (
              <Button
                type='submit'
                variant='contained'
                color='primary'
                className={classes.button}
              >
                Next
              </Button>
            )}
              
            </div>
          </Form>
        )}
      </Formik>
    </>
  );
};

FirstStep.propTypes = {
  formData: PropTypes.object.isRequired,
  setFormData: PropTypes.func.isRequired,
  nextStep: PropTypes.func.isRequired
};
