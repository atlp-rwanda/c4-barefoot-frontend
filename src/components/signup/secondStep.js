import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { Formik, Form, Field } from 'formik';
import { TextField, Select, InputAdornment, CircularProgress, MenuItem, FormControl, InputLabel } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import { Button } from '@material-ui/core';
import Box from '@material-ui/core/Box';
import { AccountCircle, Language, Subject, EventBusy } from "@material-ui/icons";
import ListItemIcon from '@material-ui/core/ListItemIcon';
import axios from 'axios'
import Avatar from '@material-ui/core/Avatar';
import { useTranslation } from 'react-i18next';


const useStyles = makeStyles(theme => ({
  form: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    '@media(max-width: 320px)' : {
      display: 'grid'
    },
  },
  button: {
    margin: theme.spacing(1)
  },
  large: {
    width: theme.spacing(7),
    height: theme.spacing(7),
  },
  button: {
    margin: theme.spacing(1),
  },
  label: {
    display: 'inline-flex',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#d5d5d5',
    color: 'black',
    padding: '0.3rem',
    fontFamily: 'sans-serif',
    borderRadius: '0.3rem',
    cursor: 'pointer',
    margin: '1rem',
  },
  uploadSpinner: {
    width: '20px !important',
    height: '20px !important',
    margin: '0 5px -5px 5px'
  }
}));

export  const SecondStep = ({
  formData,
  setFormData,
  nextStep,
  prevStep
}) => {
  const { t, i18n } = useTranslation();
  const classes = useStyles();
  const [direction, setDirection] = useState('back');
  const [photo, setPhoto] = useState('');
  const [loading, setLoading] = useState(false);
  useEffect(() => { setLoading(false); setPhoto('') }, [])
  return (
    <>
      <Formik
        initialValues={formData}
        onSubmit={values => {
          setFormData(values);
          direction === 'back' ? prevStep() : nextStep();
        }}
      >
        {({ errors, touched, setFieldValue }) => (
          <Form className={classes.form}>
            <Box display='inline-flex' justifyContent='center' alignItems='center'>
              <Avatar 
              alt="user" 
              name='profile_picture' 
              margin='normal'
              src={photo ? photo : formData.profile_picture }
              className={classes.large}/>
              <input 
              id="file" 
              name="profile_picture" 
              type="file" 
              hidden={true}
              onChange={(e) => {
                const profile_picture = e.target.files[0]
                const formData = new FormData()
                formData.append('upload_preset', process.env.UPLOAD_PRESET)
                formData.append('file', profile_picture)
                setLoading(true) 
                setPhoto('')               
                axios.post(process.env.IMAGE_UPLOAD_LINK, formData)
                .then(res => {setFieldValue("profile_picture", res.data.secure_url); setLoading(false); setPhoto(res.data.secure_url)})
                .catch(err => { setLoading(false) })
              }}
              />
              
              <label htmlFor="file" className={classes.label}>
                <i className="fa fa-cloud-upload" aria-hidden="true"></i> 
                <div>
                {t("Upload Profile Image")} {loading ? <CircularProgress className={classes.uploadSpinner}/> : ''}
                </div>

              </label>
            </Box>
            <Field 
            name='bio' 
            label={t("Biography")}
            as={TextField} 
            margin='normal'
              inputprops={{
                startAdornment: (
                  <InputAdornment position="start">
                    <Subject />
                  </InputAdornment>
                ),
              }}
            />
            <Field 
            name='occupation' 
            label={t("Occupation")}
            as={TextField} 
            margin='normal'
              inputprops={{
                startAdornment: (
                  <InputAdornment position="start">
                    <EventBusy />
                  </InputAdornment>
                ),
              }}
            />
            <FormControl>
              <Field 
              name="address" 
              label={t("Address")}
              as={TextField} 
                inputprops={{
                  startAdornment: (
                    <InputAdornment position="start">
                    <EventBusy />
                  </InputAdornment>
                  ),
                }}
              />
            </FormControl>
            <FormControl>
              <InputLabel>{t("Language")}</InputLabel>
              <Field 
              as={Select} 
              name="language" 
                inputprops={{
                  startAdornment: (
                    <InputAdornment position="start">
                      <Language />
                    </InputAdornment>
                  ),
                }}
              >
                <MenuItem selected value="Kinyarwanda">{t("Kinyarwanda")}</MenuItem>
                <MenuItem value="English">{t("English")}</MenuItem>
                <MenuItem value="Franch">{t("French")}</MenuItem>
              </Field>
            </FormControl>
            
            <div>
              <Button
                type='submit'
                
                variant='contained'
                color='primary'
                className={classes.button}
                onClick={() => {
                  setDirection('back'); 
                }}
              >
                {t("Back")}
              </Button>
              <Button
                type='submit'
                id='backBtn'
                variant='contained'
                color='primary'
                className={classes.button}
                onClick={() => setDirection('forward')}
              >
                {t("Next")}
              </Button>
            </div>
          </Form>
        )}
      </Formik>
    </>
  );
};

SecondStep.propTypes = {
  formData: PropTypes.object.isRequired,
  setFormData: PropTypes.func.isRequired,
  nextStep: PropTypes.func.isRequired,
  prevStep: PropTypes.func.isRequired
};
