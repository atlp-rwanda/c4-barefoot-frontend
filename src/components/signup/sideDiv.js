import React from 'react'
import { makeStyles } from '@material-ui/core/styles';
import Skeleton from '@material-ui/lab/Skeleton';
import { Box } from '@material-ui/core';

const useStyles = makeStyles((theme) => ({
    sideNav: {
        display: 'flex',
        flexDirection: 'column',
        flexWrap: 'nowrap',
        justifyContent: 'center',
        alignContent: 'center',
        fontSize: '1.5rem',
        '@media(max-width: 740px)' : {
            width: '80%'
        }
    },
    locationImg: {
        width: '70%',
        '@media(max-width: 740px)' : {
            display: 'none',
        }
    },
    img:{
        textAlign: 'center'
    },
    imgSkeleton: {
        width: '70%',
        maxWidth: '300px',
        minHeight: '248px',
        marginLeft: 'auto',
        marginRight: 'auto',
        '@media(max-width: 740px)' : {
            display: 'none'
        }
    },
    sideText:{
        textAlign: 'center',
        color: 'dodgerblue',
        '@media(max-width: 740px)' : {
            fontSize: '18px',
            marginBottom: '20px'
          },
          '@media(max-width: 360px)' : {
            marginBottom: '0'
          }
    },
    textBox: {
        height: '25px',
        marginLeft: 'auto',
        marginRight: 'auto',
        marginBottom: '0.5rem',
        width: '100%',
        maxWidth: '300px'
      },
  }));
function SideDiv({loading}){
    const classes = useStyles();
    return(
        <div className={classes.sideNav}>
            { loading ? (
                <Box className={classes.box}>
                    <Skeleton variant="rect" className={classes.textBox}/>
                    <Skeleton variant="rect" className={classes.textBox} />
                </Box>
            ) : (
                <div className={classes.sideText}>
                    <p>Please create account to get </p>
                    <p>started with Barefoot Nomard.</p>
                </div>
            )
            }
            {loading ? (
                <Box className={classes.box}>
                    <Skeleton variant="rect" className={classes.imgSkeleton}/>
                </Box>
            ) : (
                <div className={classes.img}>
                    <img className={classes.locationImg} src={'https://res.cloudinary.com/barefoot-nomad-app/image/upload/v1609407601/images/location_goruvq.png'} alt='Barefoot Nomards'/>
                </div>
            )}
        </div>
    )
}

export default SideDiv