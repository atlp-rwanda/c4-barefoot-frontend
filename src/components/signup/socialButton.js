import React, {Component} from 'react'
import FacebookLoginWithButton from 'react-facebook-login';
import GoogleLoginButton from 'react-google-login'


class SocialButtons extends Component{
    render(){
        return(
            
            (this.props.stepIndex == 0 ? (
                <>
                    <p>Or</p>
                    
                    <FacebookLoginWithButton
                        textButton='Signup with facebook'
                        size= 'small'
                        fields="name,email,picture"
                        icon="fa-facebook"/>
                    <p>Or</p>
                    <GoogleLoginButton
                        buttonText="SIGNUP WITH GOOGLE"
                        cookiePolicy={'single_host_origin'}
                    />
                </>
            ): '')
        )
    }
}

export default SocialButtons