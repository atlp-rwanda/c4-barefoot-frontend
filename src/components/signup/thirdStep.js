import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux'
import { makeStyles } from '@material-ui/core/styles';
import { Button } from '@material-ui/core';
import { List, ListItem, ListItemText } from '@material-ui/core/';
import { AccountCircle, Email, BusinessCenter, Language, Subject, EventBusy } from "@material-ui/icons";
import ListItemIcon from '@material-ui/core/ListItemIcon';
import Avatar from '@material-ui/core/Avatar';
import Snackbar from '@material-ui/core/Snackbar';
import MuiAlert from '@material-ui/lab/Alert';
import Loader from '../Loader'
import { requestSignup } from '../../redux/actions/signupRequestAction'
import { useDispatch, useSelector } from 'react-redux'
import { closeSnackbar } from '../../redux/actions/signupRequestAction';
import { useTranslation } from 'react-i18next';

const useStyles = makeStyles(theme => ({
  container: {
    textAlign: 'left'
  },
  textCenter: {
    textAlign: 'center'
  },
  button: {
    margin: theme.spacing(1)
  },
  large: {
    width: theme.spacing(10),
    height: theme.spacing(10),
  }
}));


export const Confirm = ({ formData, prevStep, nextStep }) => {
  const { t, i18n } = useTranslation();
  const classes = useStyles();
  const { first_name, last_name, email, username, occupation, bio, address, language, profile_picture} = formData;

  const dispatch = useDispatch();
  const requestingState = useSelector(state => state.signup.requesting);
  const openError = useSelector(state => state.signup.errorOpen);
  const errorMsg = useSelector(state => state.signup.error);

  const signupRequest = () => {
    const{confirmPassword, ...user} = {...formData}
    dispatch(requestSignup(user, nextStep))
  }
  const handleClose = () => {
    dispatch(closeSnackbar());
  };
  function Alert(props) {
    return <MuiAlert elevation={6} variant="filled" {...props}  itemID='alert' />;
  }
  return (
    <>
      <Loader open={requestingState} />
      <Snackbar open={openError} autoHideDuration={6000} onClose={handleClose} >
        <Alert onClose={handleClose} severity="error" >
          Error: {errorMsg ? JSON.stringify(errorMsg).replace(/[\\'"]+/g, '') : 'Error Not set'}
        </Alert>
      </Snackbar>
      <div>
        <List>
        <ListItem>
            <Avatar alt="user" name='file' src={profile_picture ? profile_picture : ''} className={classes.large}/>
          </ListItem>
          <ListItem>
            <ListItemIcon>
              <AccountCircle />
            </ListItemIcon>
            <ListItemText
              primary={t('Names')}
              secondary={first_name + ' ' + last_name}
            />
          </ListItem>
          <ListItem>
            <ListItemIcon>
              <AccountCircle />
            </ListItemIcon>
            <ListItemText
              primary={t('Username')}
              secondary={username}
            />
          </ListItem>
          <ListItem>
            <ListItemIcon>
              <Email/>
            </ListItemIcon>
            <ListItemText
              primary={t('Email')}
              secondary={email}
              
            />
          </ListItem>
          <ListItem>
            <ListItemIcon>
              <AccountCircle />
            </ListItemIcon>
            <ListItemText
              primary={t('Occupation')}
              secondary={occupation}
              
            />
          </ListItem>
          <ListItem>
            <ListItemIcon>
              <BusinessCenter />
            </ListItemIcon>
            <ListItemText
              primary={t('Address')}
              secondary={address}
              
            />
          </ListItem>
          <ListItem>
            <ListItemIcon>
              <Language />
            </ListItemIcon>
            <ListItemText
              primary={t('Language')}
              secondary={language}
              
            />
          </ListItem>
          <ListItem>
            <ListItemIcon>
              <Subject />
            </ListItemIcon>
            <ListItemText
              primary={t('Biography')}
              secondary={bio}
            />
          </ListItem>
          <ListItem>
            <ListItemIcon>
              <EventBusy />
            </ListItemIcon>
            <ListItemText
              primary={t('Occupation')}
              secondary={occupation}
            />
          </ListItem>
        </List>
        <div className={classes.textCenter}>
          <Button
            color='secondary'
            variant='contained'
            id="backBtn"
            className={classes.button}
            onClick={() => prevStep()}
          >
            {t("Back")}
          </Button>

          <Button
            color='primary'
            variant='contained'
            btn='submitBtn'
            className={classes.button}
            onClick={() => signupRequest()}
          >
            {t("Submit")}
          </Button>
        </div>
      </div>
    </>
  );
};

Confirm.propTypes = {
  formData: PropTypes.object.isRequired,
  prevStep: PropTypes.func.isRequired,
  nextStep: PropTypes.func.isRequired
};

const mapStateToProps = state => ({
  requestSignup: state.signup,
})

export default connect(mapStateToProps, { requestSignup })(Confirm)