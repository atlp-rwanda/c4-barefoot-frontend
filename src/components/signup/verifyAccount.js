import { Box, Button, makeStyles } from '@material-ui/core'
import Devider from '@material-ui/core/Divider'
import { CheckCircle } from '@material-ui/icons'
import Axios from 'axios'
import React, { useState } from 'react'
import queryString from 'query-string';
import { Skeleton } from '@material-ui/lab'

const useStyle = makeStyles(theme => ({
    container: {
        display: 'grid',
        justifyContent: 'center',
        flexDirection: 'column',
        alignItems: 'center',
        height: '400px',
        padding: '10px'
    },
    boxContainer: {
        textAlign: 'center'
    },
    box: {
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        padding: '10px',
        color: '#90a4ae'
    },
    hr: {
        height: '2px',
        margin: '5px -10px',
        '@media(max-width: 740px)': {
            margin: '5px'
        }
    },
    tokenMissing: {
        fontSize: '5vh',
        padding: '20px',
        marginBottom: '10px',
        borderBottom: '2px black solid',
    }
}))
const verify = (token, verified, setVerified ,setError) => {
    Axios.patch(`https://barefoot-nomad-app-v1.herokuapp.com/api/v1/user/verification?token=` + token)
    .then(res => {
      setVerified(true)
    })
    .catch(err => { 
      if (err.response){
        setError(err.response.data.error)
      }else if(err.request){
        setError(err.request)
      }else if(err.message){
        setError(err.message)
      }
    })
}
const verifyAccount = () => {
    const classes = useStyle()
    const [verified, setVerified] = useState(false);
    const [error, setError] = useState('');
    const parsed = queryString.parse(location.search);
    const [checked, setChecked] = useState(false)
    const token = parsed.token
    
    if(!checked){
        verify(token, verified, setVerified, setError)
        setChecked(true)
    }
    
    return (
        
        <div className={classes.container}>
            <Box className={classes.boxContainer}>
                {token && verified ? (
                <>
                    <Box className={classes.box}>
                        <CheckCircle style={{color: '#1b5e20'}}/>Your e-mail has been verified
                    </Box>
                    <Devider flexItem className={classes.hr}/>
                    <Box className={classes.box}>
                        Thank you for verifying your account. You can now continue to use BarefootNomad
                    </Box>
                    
                </>) : (
                    (!error ? (
                        <Box>
                            <Skeleton variant="text" width={300} height={40}/>
                            <Skeleton variant="text" width={300} height={40}/>
                        </Box>
                    ) : (
                        <Box className={classes.tokenMissing}>
                            { JSON.stringify(error).replace(/['"]+/g, '')}
                        </Box>
                    ))
                    
                )}
                { !error && !verified ? (
                    <Skeleton variant="rect" width={70} height={50} style={{marginLeft: 'auto', marginRight: 'auto'}}/>
                ) : (
                    <Button variant="contained" color="primary" href='/'> Go Home</Button>
                )}
                
            </Box>
            
        </div>
        
        
    )
}

export default verifyAccount