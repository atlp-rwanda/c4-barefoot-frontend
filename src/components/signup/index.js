import { FirstStep } from './firstStep';
import { SecondStep } from './secondStep';
import { Confirm } from './thirdStep';
import SideNav from './sideDiv'
import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Stepper from '@material-ui/core/Stepper';
import Step from '@material-ui/core/Step';
import StepLabel from '@material-ui/core/StepLabel';
import Typography from '@material-ui/core/Typography';
import SocialButtons from '../signup/socialButton'
import { Avatar, Box } from '@material-ui/core';
import Skeleton from '@material-ui/lab/Skeleton';
import { useTranslation } from 'react-i18next';

const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    textAlign: 'center',
    float: 'right',
    backgroundColor: '#fff',
  },
  rightSide: {
    border: '2px solid lightblue',
    padding: '1.5rem',
    '@media(max-width: 740px)' : {
      maxWidth: '450px',
    },
    '@media(max-width: 370px)' : {
      border: 'none',
      padding: 'unset'
    },
  },
  h3:{
    color: 'dodgerblue',
    margin: '1em',
    '@media(max-width: 740px)' : {
      fontSize: '15px'
    }
  },
  container: {
    display: 'grid',
    fontFamily: "'Poppins', sans-serif",
    '@media(max-width: 740px)' : {
      display: 'flex',
      flexDirection: 'column'
    },
    gridTemplateColumns: '1fr 1fr', 
    alignItems: 'center',
    justifyContent: 'center',
    padding: '1rem',
    boxSizing: 'border-box',
    marginBottom: '100px'
  },
  backButton: {
    marginRight: theme.spacing(1),
  },
  instructions: {
    marginTop: theme.spacing(1),
    marginBottom: theme.spacing(1),
  },
  textBox: {
    height: '20px',
    marginLeft: 'auto',
    marginRight: 'auto',
    marginBottom: '0.5rem',
    width: '100%',
    maxWidth: '200px'
  },
  box: {
    display: 'grid',
    alignItems: 'center'
  },
  stepperContainer: {
    display: 'flex',
    justifyContent: 'center',
    width: '400px',
    marginTop: '4px',
    
    '@media(max-width: 800px)' : {
      width: '330px'
    },
    '@media(max-width: 740px)' : {
      width: 'unset'
    },
  },
  stepperBox: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    minWidth: '100px'
  },
  stepperText: {
    height: '10px',
    width: '70%',
    marginTop: '5px'
  }
}));

function getSteps() {
  return ['Basic Information', 'More information', 'All information'];
}

function getStepContent(loading, stepIndex, formData, setFormData, handleNext, handleBack) {
  switch (stepIndex) {
    case 0:
      return (
        <FirstStep
          formData={formData}
          setFormData={setFormData}
          nextStep={handleNext}
          loading = {loading}
        />
      );
    case 1:
      return (
        <SecondStep
          formData={formData}
          setFormData={setFormData}
          nextStep={handleNext}
          prevStep={handleBack}
          loading = {loading}
        />
      );
    case 2:
      return (
        <Confirm formData={formData} nextStep={handleNext} prevStep={handleBack} loading = {loading} />
      );
    default:
  }
}

export default function SignUp() {
  const { t, i18n } = useTranslation();
  const [formData, setFormData] = React.useState({
    first_name: '',
    last_name: '',
    email: '',
    occupation: '',
    username: '',
    bio: '',
    address: '',
    language: '',
    password: '',
    profile_picture: ''
  });
  const classes = useStyles();
  const [activeStep, setActiveStep] = React.useState(0);
  const [loading, setLoading] = React.useState(true)
  const steps = getSteps();

  const handleNext = () => {
    setActiveStep((prevActiveStep) => prevActiveStep + 1);
  };

  const handleBack = () => {
    setActiveStep((prevActiveStep) => prevActiveStep - 1);
  };

  setTimeout(() => { setLoading(false)}, 1000);
  return (
    <div className={classes.container}>
      <SideNav loading={loading}/>
      <div className={classes.root}>
        <div className={classes.rightSide}>
          { loading ? (
            <Box className={classes.box}>
              <Skeleton variant="rect" className={classes.textBox}/>
            </Box>
          ) : (
            <h3 className={classes.h3}>{t("Signup with your credentials")}</h3>
          )}
          
          { loading ? (
            <Box className={classes.stepperContainer}>
              <Box className={classes.stepperBox}>
                <Skeleton variant="circle">
                  <Avatar/>
                </Skeleton>
                <Skeleton variant="rect" className={classes.stepperText}/>
                <Skeleton variant="rect" className={classes.stepperText}/>
              </Box>
              <Box className={classes.stepperBox}>
                <Skeleton variant="circle">
                  <Avatar/>
                </Skeleton>
                <Skeleton variant="rect" className={classes.stepperText}/>
                <Skeleton variant="rect" className={classes.stepperText}/>
              </Box>
              <Box className={classes.stepperBox}>
                <Skeleton variant="circle">
                  <Avatar/>
                </Skeleton>
                <Skeleton variant="rect" className={classes.stepperText}/>
                <Skeleton variant="rect" className={classes.stepperText}/>
              </Box>
            </Box>
          ) : (
            <Stepper activeStep={activeStep} alternativeLabel>
              {steps.map((label) => (
                <Step key={label}>
                  <StepLabel>{label}</StepLabel>
                </Step>
              ))}
            </Stepper>
          )}
          
          <div>
            {activeStep === steps.length ? (
              <div>
                <Typography className={classes.instructions}>Well Done <i class="fa fa-thumbs-up"></i></Typography>
                <Typography className={classes.instructions}>Now, Check your inbox <p>to verify your account</p></Typography>
              </div>
            ) : (
              <div>
                <Typography className={classes.instructions}>{getStepContent(loading, activeStep, formData, setFormData, handleNext, handleBack)}</Typography>
              </div>
            )}
          </div>
        </div>
        <SocialButtons stepIndex={activeStep} loading={loading}/>
      </div>
    </div>
  );
}

