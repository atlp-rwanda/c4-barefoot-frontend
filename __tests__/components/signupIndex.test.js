import signup from '../../src/components/views/Signup'
import React from 'react';
import toJson from "enzyme-to-json";
import { shallow } from 'enzyme';

describe('Signup index', () => {
    it('renders correctly first step', () => {
        const wrapper = shallow(<signup />)
        expect(toJson(wrapper)).toMatchSnapshot();
    });
})
