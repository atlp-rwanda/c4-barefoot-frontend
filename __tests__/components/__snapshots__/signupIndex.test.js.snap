// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup index renders correctly first step 1`] = `<signup />`;
